File: aoc-2023/src/test/java/com/capital7software/aoc/aoc2023/days/Day05Test.java
package com.capital7software.aoc.aoc2023.days;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

class Day05Test extends AdventOfCodeTestBase {
  private static final Logger LOGGER = LoggerFactory.getLogger(Day05Test.class);

  @BeforeEach
  void setUp() {
    var instance = new Day05();
    setupFromFile(instance.getDefaultInputFilename());
  }

  @Test
  void getLowestLocationNumber() {
    var instance = new Day05();
    int expected = 35;
    var actual = instance.getLowestLocationNumber(lines);
    assertEquals(expected, actual, "The lowest location number "
        + "is not what was expected: " + expected);
  }

  @Test
  void getTotalNumberOfCards() {
    var instance = new Day05();
    int expected = 46;
    var actual = instance.getLowestLocationNumberForAnyInitialSeed(lines);
    assertEquals(expected, actual, "The lowest location number for any initial seed "
        + "is not what was expected: " + expected);
  }

  @Override
  protected Logger getLogger() {
    return LOGGER;
  }
}

File: aoc-2017/src/test/kotlin/com/capital7software/aoc/aoc2017aoc/days/Day05Test.kt
package com.capital7software.aoc.aoc2017aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day05Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day05Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day05()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testStepsToExitLoop() {
    val instance = Day05()
    val expected = 5
    val actual = instance.stepsToExitLoop(lines)

    Assertions.assertEquals(
        expected, actual, "The number of steps to take " +
        "is not what was expected: $expected"
    )
  }

  @Test
  fun testStepsToExitLoopNewRules() {
    val instance = Day05()
    val expected = 10
    val actual = instance.stepsToExitLoop(lines, true)

    Assertions.assertEquals(
        expected, actual, "The number of steps to take " +
        "is not what was expected: $expected"
    )
  }
}
